code/modules/food_and_drinks/kitchen_machinery/smartfridge.dm

```diff
@@ -207,9 +207,13 @@
 
 	var/listofitems = list()
 	for (var/I in src)
+		// We do not vend our own components.
+		if(I in component_parts)
+			continue
+
 		var/atom/movable/O = I
 		if (!QDELETED(O))
-			var/md5name = rustg_hash_string(RUSTG_HASH_MD5, O.name)				// This needs to happen because of a bug in a TGUI component, https://github.com/ractivejs/ractive/issues/744
+			var/md5name = rustg_hash_string(RUSTG_HASH_MD5, O.name) // This needs to happen because of a bug in a TGUI component, https://github.com/ractivejs/ractive/issues/744
```

Comment:
this comment is totally inaccurate, TGUI doesn't even use ractivejs anymore, it must be an ancient comment.

Reply:
removed in 767bd3c